proxy-server/options/options.js

```diff
@@ -0,0 +1,56 @@
+const proxyServerAddressInput = document.querySelector("#proxyServerAddressInput");
+const proxyServerPortInput = document.querySelector("#proxyServerPortInput");
+const proxyServerUsernameInput = document.querySelector("#proxyServerUsernameInput");
+const proxyServerPasswordInput = document.querySelector("#proxyServerPasswordInput");
+const proxyServerTypeInput = document.querySelector('input[name=proxyServerTypeInput]:checked');
+
+// Store the currently selected settings using browser.storage.local.
+function storeSettings() {
+  let proxyServerInfo = {
+		proxyServerAddress: proxyServerAddressInput.value,
+		proxyServerPort: proxyServerPortInput.value,
+		proxyServerType: proxyServerTypeInput.value
+	};
+
+	// authCrendentials object is handed directly to browser.webRequest.onAuthRequired()
+	// so it's variable name format varies from our other variables in order to fulfill the API specifcation
+	let authCredentials = {
+		username: proxyServerUsernameInput.value,
+		password: proxyServerPasswordInput.value,
+	};
+
+	console.log("Storing settings");
+  browser.storage.local.set({
+    proxyServerInfo, authCredentials
```

Comment:
nit: the indentation seems inconsistent, most of the other examples have gone for 2 spaces. This seems to jump around between 2 spaces and 8. Also that's what you get when python developers review things.